utils_linux.go

```diff
@@ -240,7 +241,15 @@ func (r *runner) run(config *specs.Process) (int, error) {
 			return -1, err
 		}
 	}
-	if r.detach {
+	if !r.create {
+		if err := process.Signal(syscall.SIGCONT); err != nil {
+			r.terminate(process)
+			r.destroy()
+			tty.Close()
+			return -1, err
+		}
+	}
+	if r.detach || r.create {
```

Comment:
On Tue, May 17, 2016 at 11:25:06AM -0700, Michael Crosby wrote:
> > +	if r.detach || r.create {
>
> When create returns without an error then the container is created
> and ready to be started.  You cannot attach because create cannot
> return.

But somebody needs to stay around to collect the container exit code,
tear down cgroups, de-register the state, etc., right?  See waitpid
notes in [1].  In the ccon implementation the host-side ‘create’
process stays around to reap the container process [2], and folks who
want to get their terminal back to run ‘start’ can background the
‘create’ process [3].

[1]: https://github.com/opencontainers/runtime-spec/pull/384#issuecomment-213673145
[2]: https://github.com/wking/ccon/tree/create-start#lifecycle
[3]: https://github.com/opencontainers/runc/pull/827/#discussion_r63464781